src/Microsoft.Health.Dicom.Core/Features/Update/UpdateRequestValidator.cs

```diff
@@ -0,0 +1,94 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using EnsureThat;
+using FellowOakDicom;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.Validation;
+using Microsoft.Health.Dicom.Core.Messages.Update;
+using Microsoft.Health.Dicom.Core.Models.Update;
+
+namespace Microsoft.Health.Dicom.Core.Features.Update;
+
+/// <summary>
+/// Provides functionality to validate an <see cref="UpdateInstanceRequest"/>.
+/// </summary>
+public static class UpdateRequestValidator
+{
+    private const int MaxStudyInstanceUids = 50;
```

Comment:
nit: We can move this to UpdateTags. UpdateTags class could be like `QueryLimit`, where it contains the filters and constraints